File: code/787/solution.cpp
#include <iostream>
#include <map>
#include <math.h>
using namespace std;
const int N = 1e9;
map<int, long long> m;
long long cal(long long n) {
    if (m[n])
        return m[n];
    int z         = sqrt(n);
    long long ret = n * n / 8;
    for (int g = 3; g <= z; g += 2) {
        ret -= cal(n / g);
    }
    for (int x = 1; x <= n / (z + 1); x++) {
        ret -= ((n / x + 1) / 2 - (n / (x + 1) + 1) / 2) * cal(x);
    }
    m[n] = ret;
    return ret;
}
int main() {
    cout << 1 + 2 * cal(N) << endl;
    return 0;
}

File: code/787/solution.go
package main

import (
	"fmt"
	"math"
)

var m = map[int64]int64{}

func F(N int64) int64 {
	if m[N] > 0 {
		return m[N]
	}
	s := N * N / 8
	z := int64(math.Sqrt(float64(N)))
	for g := int64(3); g <= z; g += 2 {
		s -= F(N / g)
	}
	for x := int64(1); x <= N/(z+1); x++ {
		s -= ((N/x+1)/2 - (N/(x+1)+1)/2) * F(x)
	}
	m[N] = s
	return s
}

func main() {
	fmt.Println(1 + 2*F(1000000000))
}
